lib/AST/Type.cpp

```diff
@@ -3362,6 +3362,11 @@ Type ArchetypeType::getExistentialType() const {
   auto constraint = ProtocolCompositionType::get(
      ctx, constraintTypes, requiresClass());
 
+  // If the archetype is only constrained to a class type,
+  // return the class type directly.
+  if (!constraint->isConstraintType())
+    return constraint;
```

Comment:
Maybe assert that it's a class type here since that should be the only other possibility?

Reply:
Done 👍

Reply:
Sorry, it could also be a BoundGenericClassType. You can do `assert(constraint->getClassOrBoundGenericClass())`